lib/avro/builder/type_factory.rb

```diff
@@ -10,6 +10,16 @@ module TypeFactory
 
       private
 
+      def type_dispatch(avro_type_or_name, namespace = nil)
```

Comment:
I was a bit confused by this method name. It wasn't clear at the calling sites what it was doing (especially when the block would kick in). Is there a better name to describe what it's doing (or at a minimum we should add some method level comments)?